DRAFT: PWA: intro to basic tutorial (#25490)
mdn/content: files/en-us/web/progressive_web_apps/tutorials/intro/index.md

---
title: Creating your first PWA

> slug: Web/Progressive_web_apps/tutorials/intro

Reviewer: Drive-by comment
```suggestion
slug: Web/Progressive_web_apps/Tutorials/Intro
```

---

Progressive web applications, known as PWAs, are a way to build applications for all devices using the web technologies we all know and love: HTML, CSS, and JavaScript.

> ## The benefit of PWA

Reviewer 2: ```suggestion
## The benefit of PWAs
```

Using the languages of the web, we can create fully functioning applications that work both online and offline, both in the browsers and on the user's operating systems.

### A single code base installable everywhere

Historically, operating system specific applications needed to be written in operating system targeted programming languages, then compiled and packaged based on OS language and application requirement, which differ by operating system, and distributed to users thru downloads, often in proprietary app stores, and then installed by the user.

For an application to be installable on an operating system, such as Windows, iOS, MacOS, Linux, and Android, the application had to be developed in a programming languages, like C#, .Net, Objective C, Swift, Kotlin, Java, or Python. PWAs solve this problem: they use a single code base, written using regular open web technologies.

> Historically, in most cases, the application code had to be compiled and packaged using the operating system's SDK into downloadable .exe, .dmg, or other installable files or were made available through a vendor like the Apple App Store, [Goole Play](https://play.google.com/store/apps), [Microsoft Store App](https://apps.microsoft.com/store/apps), or similar. PWAs don't have these requirements.

Reviewer 2: ```suggestion
Historically, in most cases, the application code had to be compiled and packaged using the operating system's SDK into downloadable .exe, .dmg, or other installable files or were made available through a vendor like the Apple App Store, [Google Play](https://play.google.com/store/apps), [Microsoft Store App](https://apps.microsoft.com/store/apps), or similar. PWAs don't have these requirements.
```

> Like regular web sites, PWA are hosted on and downloable from web servers. They don't require any packaging. They can be seamlessly installed on any operating system without the need of app stores (nor app store approval and fees). PWA don't require any additional programming language knowledge, any proprietary SDK, or even an app store.

Reviewer 2: ```suggestion
Like regular web sites, PWAs are hosted on and downloadable from web servers. They don't require any packaging. They can be seamlessly installed on any operating system without the need of app stores (nor app store approval and fees). PWAs don't require any additional programming language knowledge, any proprietary SDK, or even an app store.
```

While PWAs don't require an intermediary, the Play store and App Store do distribute PWAs, undifferentiated from other iOS or Android apps.

### Regular websites that are progressively enhanced

PWAs are a way of progressively enhancing websites to make them available on a user's device, even offline. By including a manifest file and service worker, and serving the website over SSL, any website can become a PWA.

By default, PWAs are regular websites. They use the same technologies as regular websites. They are also linkable, discoverable via search engines, and visible in a browser, just like regular websites.

What makes a PWA different from a regular web applications is that they are progressively enhanced to be able to be installed.

### Similarities to installed applications

Once installed, PWAs can be made to appear and act similar to other applications installed on the user's device:

- Application window

  - : Similar to other installed applications, depending on PWA settings defined by the developer and set with user preferences, the PWA can be made to open in its own application window.

- Application icon

>   - : PWA display an application icon in the same location as other installed applications installed on the users' operating system: this can be an icon on the homescreen, in the tool bar, in the applications folder, or wherever the devics displays the icons of available applications.

Reviewer 2: ```suggestion
  - : PWAs display an application icon in the same location as other installed applications installed on the users' operating system: this can be an icon on the homescreen, in the tool bar, in the applications folder, or wherever the device displays the icons of available applications.
```
